### DISCUSSION TITLE
Adding keyword arguments to a variadic function?

### DISCUSSION THREAD

**Discussion by iamkisly** [dotnet/csharplang#6346] [Language Ideas]
Hello. Has the possibility of extending the functionality of the ```params``` keyword from the variadic function to a full-fledged keyword arguments, as it is done in Python with their ```**kwargs```, been considered before?

```python
def kwargs_test(**kwargs):
    print(kwargs)

kwargs_test(hello='world', jeffrey='Richter')
'''{'hello': 'world', 'jeffrey': 'Richter'}'''
```

It would be nice to be able to dotnet wrap named arguments in a ```Dictionary <string, object>``` :
```csharp

public dotnet_kwargs(params Dictionary<string, string> kwargs)
{
    // do something like
    Console.WriteLine(JsonSerializer.Serialize(kwargs));
}

dotnet_kwargs(hello: 'world', jeffrey: 'Richter')
// {"hello":"world","jeffrey":"Richter"}
```
What do you think about this ?

**Comment by CyrusNajmabadi** [comment]
Being able to simply make dictionaries is part of this work: https://github.com/dotnet/csharplang/issues/5354

**Reply by iamkisly** [reply]
Yes, I've seen this discussion and so my question is specifically about wrapping named arguments.

**Reply by CyrusNajmabadi** [reply]
i don't believe we'd do that given that you could just trivially make a dictionary just by adding something like `[]` around it.  It would be a lot of language complexity for little gain at that point.  But we can always see how the community as a whole feels :)

**Reply by iamkisly** [reply]
Well .. perhaps this can be done using a source generator. Need to experiment :)

**Comment by ufcpp** [comment]
https://github.com/dotnet/csharplang/issues/1333
